### PR TITLE
New post feature

### PR DESCRIPTION
**Pull request by ryanturnbull**
Again - not worrying too much about the theme at this point 😄 will need to go over that completely a bit later

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 @roam:registry=http://registry.runsonsauce.com:3000/
-//registry.runsonsauce.com:3000/:_authToken=${NPM_TOKEN}
+//registry.runsonsauce.com:3000/:_authToken=77c0819e-7e0d-4723-9b1f-a31258615d03
```
Comment (francosioquim) [inline .npmrc]: don't share your  token... lol
this might not work in circleCI later on

Comment (francosioquim) [reply .npmrc]: can we add this to gitignore then push?
so you can freely do whatever we want with .npmrc on your local

File: src/components/atoms/StyleButton.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  },
+  active: {
+    backgroundColor: '#ccc',
+    '&:hover': {
+      backgroundColor: '#bbb',
+    }
+  }
+})
+
+class StyleButton extends React.Component {
```
Comment (francosioquim) [inline src/components/atoms/StyleButton.js]: This looks like a **React.PureComponent** to me. What do you think?

File: src/components/organisms/NewPostForm/NewPostForm.js [hunk 3]
```diff
@@ -58,6 +76,23 @@ const NewPostForm = ({
...
+          <TextField
+            id="description"
+            name="description"
+            label="Post Description ( To appear in-app )"
+            value={values.description}
+            onChange={handleFieldChange}
+            margin="normal"
+            error={hasFieldError('description')}
+            inputProps={{
+              maxLength: 256,
```
Comment (francosioquim) [inline src/components/organisms/NewPostForm/NewPostForm.js]: We might need to set the value for this depends on sauce maxLength for description. Not required to do anything but just keep in mind.

File: src/templates/DashboardLayout.js [hunk 4]
```diff
@@ -51,7 +50,7 @@ const DashboardLayout = ({
     <Route
       {...rest} render={matchProps => (
         <React.Fragment>
-          <PageWrapper maxWidth={1280} className={className}>
+          <div className={className}>
```
Comment (francosioquim) [inline src/templates/DashboardLayout.js]: Josh has a fix for this. You might want to check with him.

File: src/utils/validation.js [hunk 5]
```diff
@@ -20,7 +20,7 @@ export function validEmail(value) {
 }
 
 export function validUrl(value) {
-  return (value && /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_.~#?&//=]*)/i.test(value))
+  return (!value || !/https?:\/\/?[-a-zA-Z0-9@:%._~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_.~#?&//=]*)/i.test(value))
```
Comment (francosioquim) [inline src/utils/validation.js]: Good catch!

File: src/components/atoms/StyleButton.js (hunk 2 above)
Comment (ryanturnbull) [reply src/components/atoms/StyleButton.js]: Good catch

File: .npmrc (hunk 1 above)
Comment (ryanturnbull) [reply .npmrc]: Yeah good idea 👍

**Pull request merged**